### PR TITLE
Added contact form and tests

### LINKED ISSUE #200: Setup contact form

### PR DESCRIPTION
**Pull request by straleyb**
Fixes #200

### CODE REVIEW

**Comment by straleyb** [comment]
Dont merge this yet. I want to refactor the tests I wrote and clean up some stuff.

**Comment by straleyb** [comment]
rebased

File: app/controllers/contacts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      flash[:success] = "Thank you for contacting us!" if mailer.deliver_now
+      redirect_to root_path
+    end
+  end
+
+  private
+
+  def mailer
+    if Setting.contact.blank?
+      contact_mailer.contact_email(@email, @message, I18n.t('mailer.default_email'))
```
Comment (terrellt) [inline app/controllers/contacts_controller.rb]: Didn't we have a way to set defaults for settings?

File: app/views/shared/_footer.html.erb [hunk 2]
```diff
@@ -15,8 +15,8 @@
     <div class="black row links">
       <div class="col-xs-12 col-md-offset-4 col-md-4">
         <p class="left">
-          <a href="https://oregondigital.org/contact"> Contact Us </a> | 
-          <a href="https://oregondigital.org/copyright"> Copyright </a> | 
+          <a href="<%= contact_us_path %>"> Contact Us </a> | 
+          <a href="https://oregondigital.org/copyright"> Copyrights </a> |
```
Comment (terrellt) [inline app/views/shared/_footer.html.erb]: Why would you change this back?

Comment (straleyb) [inline app/views/shared/_footer.html.erb]: I think that got transposed during the rebase. Ill change it.

File: app/controllers/contacts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+class ContactsController < ApplicationController
+
+  def index
+  end
+  
+  def create
+    @email ||= contact_params["email"]
+    @message = contact_params["message"]
+    if @email.blank?
+      flash[:error] = "Please enter an email address and try again"
```
Comment (terrellt) [inline app/controllers/contacts_controller.rb]: I18n

File: app/controllers/contacts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      render :index
+    else
+      flash[:success] = "Thank you for contacting us!" if mailer.deliver_now
```
Comment (terrellt) [inline app/controllers/contacts_controller.rb]: I18n.

File: app/controllers/contacts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      contact_mailer.contact_email(@email, @message, Setting.contact) 
+    end
+  end
+
+  def contact_mailer
+    ContactMailer
+  end
+
+  def contact_params
+    params["Contact"]
```
Comment (terrellt) [inline app/controllers/contacts_controller.rb]: Strong params?

File: app/controllers/contacts_controller.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
+class ContactsController < ApplicationController
+
+  def index
+  end
+  
+  def create
+    @email ||= contact_params["email"]
```
Comment (terrellt) [inline app/controllers/contacts_controller.rb]: Prefer symbols to strings for param hash access.

File: app/controllers/contacts_controller.rb (hunk 1 above)
Comment (straleyb) [inline app/controllers/contacts_controller.rb]: Not that I remember. From what I remember, we we made settings callable by the name of the setting using method missing. Thats all I remember though.

Comment (straleyb) [inline app/controllers/contacts_controller.rb]: I found it. You definitely can. And that explains why it was returning an empty string.

**Pull request merged**